### PR TITLE
fix: allow any mark config in style config

### LINKED ISSUE #5988: binSpacing missing from bar style
This is invalid but used in vega-themes. We cannot update. 

```
   style: {
    bar: {
      binSpacing: 2
    }
```

### PR DESCRIPTION
**Pull request by domoritz**
fixes #5988

### CODE REVIEW

**Comment by domoritz** [comment]
I'm working on making the generated schema nicer.

File: build/vega-lite-schema.json [hunk 1]
```diff
@@ -124,6 +124,166 @@
...
       ]
     },
+    "AnyMarkConfig": {
+      "anyOf": [
+        {
+          "anyOf": [
+            {
+              "$ref": "#/definitions/AreaConfig"
+            },
+            {
```
Comment (kanitw) [inline build/vega-lite-schema.json]: Why are there these "not"s?  Kinda weird.

Comment (domoritz) [reply build/vega-lite-schema.json]: https://github.com/vega/ts-json-schema-generator/issues/344. It's pretty hairy and I'll solve it later.

**Comment by domoritz** [comment]
I simplified the schema. Should be good now.

**Pull request merged**